std-bits/base/src/main/java/org/enso/base/parser/Separators.java

```diff
@@ -0,0 +1,131 @@
+package org.enso.base.parser;
+
+import static org.enso.base.parser.NumberWithSeparators.isDigit;
+
+import java.nio.CharBuffer;
+
+public record Separators(char first, char second, int count, int endIdx, int lastSeparator) {
+  /** Check if the character is a separator. */
+  static boolean isSeparator(char c) {
+    return c == '.' || c == ',' || c == ' ' || c == '\'';
+  }
+
+  /**
+   * Strip out the specified separators and replace with just full stop for decimal. If any
+   * character other than a digit, thousands or decimal separator is encountered then return null.
+   * If multiple decimal separators are encountered then return null.
+   */
+  static CharSequence strip(
+      CharSequence value, int startIdx, int endIdx, char thousands, char decimal) {
+    boolean foundDecimal = false;
+    char[] results = new char[endIdx - startIdx];
+    int resultIdx = 0;
+    for (int i = startIdx; i < endIdx; i++) {
+      char c = value.charAt(i);
+      if (c == decimal) {
+        if (foundDecimal) {
+          return null;
+        }
+        results[resultIdx++] = '.';
+        foundDecimal = true;
+      } else if (isDigit(c)) {
+        results[resultIdx++] = c;
+      } else if (c != thousands) {
+        return null;
+      }
+    }
+    return CharBuffer.wrap(results, 0, resultIdx);
+  }
+
+  private static boolean validChar(char c, char first, char second) {
+    return isDigit(c)
+        || (first == NumberWithSeparators.Constants.NONE
+            ? isSeparator(c)
+            : (second == NumberWithSeparators.Constants.NONE
+                ? c == first || c == ',' || c == '.'
+                : c == second));
+  }
+
+  /**
+   * Find the number and separators section. Validate the spacing of separators. Return the
+   * separators found or null if invalid.
+   */
```

Comment:
Do I understand correctly that this only finds up to 2 separator characters, but does not yet validate them?

i.e. if I pass it `123,456.789,123,456,111.222` it will still parse it fine and it will be rejected at a later stage?

Reply:
No, it will read up to the end of the valid number (`123,456.789`). 
It will then fail when the parser next detects a number.